pkg/datastore/mysql/mysql.go

```diff
@@ -82,22 +82,183 @@ func (m *MySQL) Find(ctx context.Context, kind string, opts datastore.ListOption
 
 // Get implementation for MySQL
 func (m *MySQL) Get(ctx context.Context, kind, id string, v interface{}) error {
-	return datastore.ErrUnimplemented
+	query := buildGetQuery(kind)
+	row := m.client.QueryRowContext(ctx, query, id)
+	var val string
+	err := row.Scan(&val)
+	if err == sql.ErrNoRows {
+		return datastore.ErrNotFound
+	}
+	if err != nil {
+		m.logger.Error("failed to get entity",
+			zap.String("id", id),
+			zap.String("kind", kind),
+			zap.Error(err),
+		)
+		return err
+	}
+
+	return decodeJSONValue(val, v)
 }
 
 // Create implementation for MySQL
 func (m *MySQL) Create(ctx context.Context, kind, id string, entity interface{}) error {
-	return datastore.ErrUnimplemented
+	stmt, err := m.client.PrepareContext(ctx, fmt.Sprintf("INSERT INTO %s (id, data) VALUE (UUID_TO_BIN(?,true), ?)", kind))
+	if err != nil {
+		m.logger.Error("failed to create entity",
+			zap.String("id", id),
+			zap.String("kind", kind),
+			zap.Error(err),
+		)
+		return err
+	}
+	defer stmt.Close()
+
+	val, err := encodeJSONValue(entity)
+	if err != nil {
+		m.logger.Error("failed to create entity",
+			zap.String("id", id),
+			zap.String("kind", kind),
+			zap.Error(err),
+		)
+		return err
+	}
+
+	// In case the given model is `project`, id is not in uuid type so just generate random uuid instead.
+	if _, err := uuid.Parse(id); err != nil {
+		id = uuid.New().String()
+	}
+
+	_, err = stmt.ExecContext(ctx, id, val)
+	if err != nil && err.(*mysql.MySQLError).Number == 1062 {
+		return datastore.ErrAlreadyExists
+	}
+	if err != nil {
+		m.logger.Error("failed to create entity",
+			zap.String("id", id),
+			zap.String("kind", kind),
+			zap.Error(err),
+		)
+		return err
+	}
+	return nil
 }
 
 // Put implementation for MySQL
 func (m *MySQL) Put(ctx context.Context, kind, id string, entity interface{}) error {
-	return datastore.ErrUnimplemented
+	stmt, err := m.client.PrepareContext(ctx, fmt.Sprintf("INSERT INTO %s (id, data) VALUE (UUID_TO_BIN(?,true), ?) ON DUPLICATE KEY UPDATE data = ?", kind))
```

Comment:
This statement is the same for all kinds, but I think we also could break this part to another function like `buildPutQuery`, to be more unified and can write some tests for it.